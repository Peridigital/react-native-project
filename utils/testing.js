import ShallowRenderer from 'react-test-renderer/shallow'

/**
 * Creates a shallow rendered string representation of a given react component
 *
 * @param {any} component - A React component to be snapshotted ex: <Text>Hello World<Text/>
 *
 * @returns {string} A shallow rendered string representation of the React component
 */
export function shallow (component) {
  const renderer = new ShallowRenderer()
  renderer.render(component)
  return renderer.getRenderOutput()
}
