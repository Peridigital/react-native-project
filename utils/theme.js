const colors = {
  primaryBackground: '#0C0D12',
  secondaryBackground: '#151821',
  primaryText: '#EAEAEA',
  border: '#7E4033',
  borderHighlight: '#D0A499'
}

export default colors
