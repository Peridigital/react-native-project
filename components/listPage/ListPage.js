import React from 'react'
import { noop } from 'lodash'
import {
  View,
  FlatList
} from 'react-native'
import PropTypes from 'prop-types'
import style from './ListPage.style'
import { generateListingComponent } from './ListPage.utils'

export function ListPage ({ starships, navigateToDetails, hasFontsLoaded }) {
  return (
    <View style={style.container}>
      <FlatList
          data={starships}
          renderItem={generateListingComponent(navigateToDetails, hasFontsLoaded)}
        />
    </View>
  )
}

ListPage.propTypes = {
  starships: PropTypes.array,
  navigateToDetails: PropTypes.func,
  hasFontsLoaded: PropTypes.bool
}

ListPage.defaultProps = {
  navigateToDetails: noop,
  hasFontsLoaded: false
}

export default ListPage
