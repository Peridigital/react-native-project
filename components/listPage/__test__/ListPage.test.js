/* eslint-disable no-undef */
import React from 'react'
import { shallow } from '../../../utils/testing'
import ListPage from '../ListPage'

describe('ListPage', () => {
  it('should render', () => {
    const wrapper = shallow(<ListPage />)

    expect(wrapper).toMatchSnapshot()
  })
})
