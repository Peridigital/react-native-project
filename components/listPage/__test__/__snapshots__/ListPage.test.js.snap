// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPage should render 1`] = `
<View
  style={
    {
      "backgroundColor": "#0C0D12",
      "gap": 15,
    }
  }
>
  <FlatList
    renderItem={[Function]}
  />
</View>
`;
